### ISSUE TITLE
Clarification on loss for multiclass classifiers.

### ISSUE THREAD

**Issue by agkphysics** [adriangb/scikeras#78]
I noticed that when using `sparse_categorical_crossentropy` loss in a model with `KerasClassifier`, it fails when providing numeric labels. When transforming the output `y`, `LabelEncoder` expects it to be a numeric value as opposed to the one-hot encoding created in `_post_process_y()`. Normal Keras fitting and predicting *will* work in this case.

On the other hand, when the loss is `categorical_crossentropy`, providing numeric labels works becuase `_check_output_model_compatibility()` checks for `categorical_crossentropy` and makes the appropriate transformation to one-hot encoding. However, normal Keras fitting and predicting *will not* work in this case because it requires the labels to be one-hot encoded.

Example:
```python
import numpy as np
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Dense
from scikeras.wrappers import KerasClassifier


def build_model():
    model = Sequential()
    model.add(Dense(5, activation='softmax', input_shape=(20,)))
    model.compile(optimizer='adam', loss='categorical_crossentropy')
    return model


X = np.random.default_rng().random((100, 20))
y = np.random.default_rng().integers(5, size=(100,))

clf = KerasClassifier(build_model)
clf.fit(X, y)
y_pred = clf.predict(X)  # This will fail when using 'sparse_categorical_crossentropy'
print(y_pred.shape)

model = build_model()
model.fit(X, y)  # This will fail when using 'categorical_crossentropy'
model.predict(X)
y_pred = model.predict(X)
print(y_pred.shape)
```

This isn't a bug so much as something that perhaps needs clarification that you should use `categorical_crossentropy`, or have an additional check in `_check_output_model_compatibility()`.

**Comment by adriangb** [comment]
Thank you for reporting this.

I think that we need to: better organize the expected behavior given each input-type, loss-type and number of outputs. It would help to
(1) Have a table of expected results (maybe 1 table for single-outputs and 1 for multi-output).
(2) Move all of the logic inside `KerasClassifier.{pre, post}_process_y` and `_check_output_model_compatibility` into an expanded version of `scikeras._utils.LabelDimensionTransformer` and make it public. That way the logic and state is contained in a single place instead of scattered around `scikeras.wrappers`.

**Comment by adriangb** [comment]
@agkphysics to be clear and explicitly answer your question/report, I'd consider this a bug since SciKeras should _always_ allow you to use the same inputs you used with Keras. SciKeras via KerasClassifier should be able to automatically wrangle your input data into the right format which is what allows your example using `categorical_crossentropy` to succeed with SciKeras even when it fails with Keras.

**Comment by agkphysics** [comment]
@adriangb what you said sounds good. As I said, it's not a big deal at the moment; I can simply change the loss as necessary.

**Comment by adriangb** [comment]
This should be fixed in master (and the next release) @agkphysics
